src/pages/registration/resource_type_tab/SelectRegistrationTypeField.tsx

```diff
@@ -68,19 +67,27 @@ export const SelectRegistrationTypeField = () => {
   const updateRegistrationData = (newInstanceType: PublicationInstanceType) => {
     if (newInstanceType !== currentInstanceType) {
       const newContextType = getMainRegistrationType(newInstanceType);
-      const newMainType = newContextType !== values.entityDescription?.reference?.publicationContext?.type;
+      const oldContextType = values.entityDescription?.reference?.publicationContext?.type;
+      const contextTypeIsChanged = newContextType !== oldContextType;
+
       switch (newContextType) {
         case PublicationType.PublicationInJournal:
-          newMainType &&
+          if (contextTypeIsChanged) {
+            // If we move between category groups we reset all fields
             setFieldValue(contextTypeBaseFieldName, { type: PublicationChannelType.UnconfirmedJournal }, false);
-          setFieldValue(
-            instanceTypeBaseFieldName,
-            { ...emptyJournalPublicationInstance, type: newInstanceType },
-            false
-          );
+            setFieldValue(instanceTypeBaseFieldName, { ...emptyBookPublicationInstance, type: newInstanceType }, false);
```

Comment:
Skulle denne vært emptyJournalPublicationInstance?
```suggestion
            setFieldValue(instanceTypeBaseFieldName, { ...emptyJournalPublicationInstance, type: newInstanceType }, false);
```

Reply:
Good catch! 😵‍💫